tests/scrape.py

```diff
@@ -1030,6 +1034,46 @@ def enclosing(self, tagname=None, content=None, enders=[], **attrs):
         c = content is not None and ' with content %r' % content or ''
         raise ScrapeError('no %s found%s%s' % (tag, a, c))
 
+class Document(Region):
+    """A document returned as an HTTP response.
+    """
+
+    def __init__(self, content_str, headers, charset=None):
```

Comment:
Please document that not passing charset will result in xpath() not working. I'd assume that the charset will be automatically determined by default.